docs/Reference/Server.md

```diff
@@ -1048,6 +1049,20 @@ fastify.routing(req, res)
 Method to add routes to the server, it also has shorthand functions, check
 [here](./Routes.md).
 
+#### hasRoute
+<a id="hasRoute"></a>
+
+Method to check if a route is already registered to the internal router:
+
+```js
+fastify.hasRoute({
+  url: '/',
+  method: 'GET',
+  constraints: { version: '1.0.0' }
+})
+```
```

Comment:
What does this return? How is it used? What are the available properties of the input object? What is each of their purpose?

See lines 998-1003 above.

Reply:
better?